### PR TITLE
Ngdialog dependency and MeetingLogin error handling

### PR DESCRIPTION
**Pull request by tkenakka**
http://likeastore.github.io/ngDialog/
http://ngmodules.org/modules/ngDialog

### CODE REVIEW

**Comment by tkenakka** [comment]
Not tested on IE yet, deployed to dev server.

File: app/scripts/controllers/overview.ctrl.js [hunk 1]
```diff
@@ -79,14 +79,16 @@ angular.module('dashboard')
             }
         }
 
-        self.loginMeeting = function (meetingItem, meetingRole, personGuid) {
-            $log.debug("overviewCtrl.loginMeeting: \n - meeting:\n" +JSON.stringify(meetingItem) +"\n - role: " +JSON.stringify(meetingRole) + "\n - personGuid: " +personGuid);
+        self.loginMeeting = function loginMeetingFn(meetingItem, meetingRole, personGuid) {
```
Comment (evidentti) [inline app/scripts/controllers/overview.ctrl.js:82]: Is 'loginMeetingFn' needed?

File: app/scripts/services/ahjo.meeting.srv.js [hunk 2]
```diff
@@ -248,7 +248,7 @@ angular.module('dashboard')
             return deferred.promise;
         };
 
-        self.meetingLogin = function (meetingGuid, meetingRole, personGuid) {
+        self.meetingLogin = function meetingLoginFn(meetingGuid, meetingRole, personGuid) {
```
Comment (evidentti) [inline app/scripts/services/ahjo.meeting.srv.js:251]: Is 'meetingLoginFn' needed?

File: app/scripts/services/ahjo.meeting.srv.js [hunk 3]
```diff
@@ -269,7 +269,7 @@ angular.module('dashboard')
             return deferred.promise;
         };
 
-        self.meetingLogout = function (meetingGuid, meetingRole, personGuid) {
+        self.meetingLogout = function meetingLogoutFn(meetingGuid, meetingRole, personGuid) {
```
Comment (evidentti) [inline app/scripts/services/ahjo.meeting.srv.js:272]: Is 'meetingLogoutFn' needed?

File: app/scripts/services/dialogutils.srv.js [hunk 4]
```diff
@@ -98,13 +98,60 @@ angular.module('dashboard')
...
+
+        /*
+         * @name dashboard.dialogutils.showError
+         * @description .
+         * @param {string} Scope to use for communicating with the dialog
+         * @param {string} String id for title.
+         * @param {string} String id for body
+         * @returns {} ngDialog
+         */
+        self.showError = function showErrorFn(aTitleStrId, aBodyStrId) {
```
Comment (evidentti) [inline app/scripts/services/dialogutils.srv.js:110]: Is 'showErrorFn' needed?

File: app/scripts/services/utils.srv.js [hunk 5]
```diff
@@ -105,6 +105,70 @@ angular.module('dashboard')
                 $log.error("Utils.openNewWin: ignored due to bad url " + aUrl);
             }
         };
+        /*
+         * @name dashboard.utils.stringIdForError
+         * @description Resolves a localized string id for an operation result code.
+         * @param {number} Result code to parse.
+         * @returns {string} String id.
+         */
+        Utils.stringIdForError = function stringIdForErrorFn(aNum) {
```
Comment (evidentti) [inline app/scripts/services/utils.srv.js:114]: Is 'stringIdForErrorFn' needed?

File: app/scripts/services/utils.srv.js [hunk 6]
```diff
@@ -105,6 +105,70 @@ angular.module('dashboard')
...
+            return res;
+        };
+
+        /*
+         * @name dashboard.utils.processAhjoError
+         * @description Parses an Ahjo Meeting API response for any errors and displays a result dialog if necessary.
+         * @param {string} a1 REST response to process. Expects a HTTP response in a `status` property and an Ahjo Meeting API response in `data.error`.
+         * @returns {integer} Parsed error code, 0 if no error, -1 if bad response object.
+         */
+        Utils.processAhjoError = function processAhjoErrorFn(aResp) {
```
Comment (evidentti) [inline app/scripts/services/utils.srv.js:150]: Is 'processAhjoErrorFn' needed?

File: app/scripts/services/utils.srv.js [hunk 7]
```diff
@@ -105,6 +105,70 @@ angular.module('dashboard')
...
+                    break;
+                case CONST.MTGAPICODES.K1004.value:
+                    res = CONST.MTGAPICODES.K1004.strId;
+                    break;
+                case CONST.MTGAPICODES.K1005.value:
+                    res = CONST.MTGAPICODES.K1005.strId;
+                    break;
+
+                default:
+                    $log.$log("Utils.stringIdForError: didn't find a string id for '" + aNum + "', defaulting to " + res);
```
Comment (evidentti) [inline app/scripts/services/utils.srv.js]: $log.$log is not a function. (In '$log.$log("Utils.stringIdForError: didn't find a string id for '" + aNum + "', defaulting to " + res)', '$log.$log' is undefined)

**Comment by evidentti** [comment]
LGTM with comments

**Comment by tkenakka** [comment]
Thanks.
Log issue fixed
Named functions are a convenience, shouldn't make a difference other than when debugging.
Merging.

**Pull request merged**